Desktop/FirmwareInstaller/FirmwareInstaller/Services/DiscoveryService.cs

```diff
@@ -1,31 +1,83 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.IO.Ports;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Management;
+using System.Text.RegularExpressions;
 
 namespace FirmwareInstaller.Services
 {
+    /// <summary>
+    /// Represents a serial device.
+    /// </summary>
+    internal class COMDevice: IComparable<COMDevice>
+    {
+        public string Port { get; set; }
+        public string Name { get; set; }
+
+        #region Constructor
+        public COMDevice() { }
+        #endregion
+
+        #region Overrides
+
+        public int CompareTo(COMDevice other)
+        {
+            int currentNumber = Int32.Parse(Port.Substring(Port.Length - 1));
```

Comment:
Com port range is 1-256 which is higher than a single digit so this will not work long term.
https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000PAhzSAG

Reply:
Was this ever resolved?